File: lint-arwaky/src-python/infrastructure/http_request_client.py
"""http_request_client — Sync HTTP provider implementation."""

from __future__ import annotations

import httpx
from ..taxonomy import ResponseData, TransportUrlVO, Timeout, ContentString
from ..contract import IHttpProviderPort


class SyncHttpProvider(IHttpProviderPort):
    """Synchronous HTTP client implementing IHttpProviderPort."""

    def get(
        self, url: TransportUrlVO, timeout: Timeout = Timeout(value=2000)
    ) -> ResponseData:
        """Performs a synchronous GET request."""
        with httpx.Client(timeout=timeout.value / 1000) as client:
            resp = client.get(url.value)
            resp.raise_for_status()
            return ResponseData(value=resp.text)

    def post(
        self,
        url: TransportUrlVO,
        data: ContentString | None = None,
        timeout: Timeout = Timeout(value=2000),
    ) -> ResponseData:
        """Performs a synchronous POST request."""
        with httpx.Client(timeout=timeout.value / 1000) as client:
            payload = data.value if data else None
            resp = client.post(url.value, content=payload)
            resp.raise_for_status()
            return ResponseData(value=resp.text)

File: lint-arwaky/src-rust/infrastructure/http_request_client.rs
/// http_request_client — Sync HTTP provider implementation (runs blocking inside async).
use crate::contract::IHttpProviderPort;
use crate::taxonomy::{ContentString, ResponseData, Timeout, TransportUrlVO};
use std::time::Duration;

pub struct SyncHttpProvider;

impl SyncHttpProvider {
    pub fn new() -> Self {
        Self
    }
}

#[async_trait::async_trait]
impl IHttpProviderPort for SyncHttpProvider {
    async fn get(&self, url: TransportUrlVO, timeout: Option<Timeout>) -> Result<ResponseData, String> {
        let dur = timeout.map(|t| Duration::from_millis(t.value as u64)).unwrap_or(Duration::from_secs(2));
        let client = reqwest::blocking::Client::builder().timeout(dur).build().map_err(|e| e.to_string())?;
        let resp = client.get(&url.value).send().map_err(|e| e.to_string())?;
        let text = resp.text().map_err(|e| e.to_string())?;
        Ok(ResponseData::new(serde_json::Value::String(text)))
    }

    async fn post(&self, url: TransportUrlVO, body: ContentString, timeout: Option<Timeout>) -> Result<ResponseData, String> {
        let dur = timeout.map(|t| Duration::from_millis(t.value as u64)).unwrap_or(Duration::from_secs(2));
        let client = reqwest::blocking::Client::builder().timeout(dur).build().map_err(|e| e.to_string())?;
        let payload = body.value.clone();
        let resp = client.post(&url.value).body(payload).send().map_err(|e| e.to_string())?;
        let text = resp.text().map_err(|e| e.to_string())?;
        Ok(ResponseData::new(serde_json::Value::String(text)))
    }
}
